src/main/scala/com/snowplowanalytics/snowplow/rdbloader/LoaderError.scala

```diff
@@ -34,6 +33,35 @@ object LoaderError {
     }
   }
 
+  /** Overloaded method to surface helpful error messages
+    * in case a [[RawConfig]] cannot be successfully
+    * transformed into a [[CliConfig]].
+    */
+  object ErrorSurface {
+    object ValidationStep extends Enumeration {
```

Comment:
Just discussed it with Rostyslav today - Scala `Enumeration` usually is a bad choice for pretty much any task people are trying to solve with them: https://medium.com/@yuriigorbylov/scala-enumerations-hell-5bdba2c1216

I even already have a function similar to `forName` mentioned in the article: https://github.com/snowplow/snowplow-rdb-loader/blob/master/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/utils/Common.scala#L149-L155

Reply:
Hey @chuwy , what are your thoughts about implementing the enumeration like so:

```
sealed trait ValidationStep
final case object LogkeyValidation extends ValidationStep
final case object FolderValidation extends ValidationStep
final case object ResolverValidation extends ValidationStep
final case object TargetValidation extends ValidationStep
```

Reply:
@dilyand yep, that's exactly right. One more small convention that I personally think is quite wide-spread is companion object for enum object, so it would be:

```scala
sealed trait ValidationStep
object ValidationStep {
  final case object LogkeyValidation extends ValidationStep
  final case object FolderValidation extends ValidationStep
  final case object ResolverValidation extends ValidationStep
  final case object TargetValidation extends ValidationStep
}
```

Most of the time you will refer to them as to `ValidationStep.SomeStep`, it makes sense to shorten it to:

```scala
sealed trait ValidationStep
object ValidationStep {
  final case object Logkey extends ValidationStep
  final case object Folder extends ValidationStep
  final case object Resolver extends ValidationStep
  final case object Target extends ValidationStep
}
```

But its pretty much my personal preference.

Reply:
I've re-implemented it without using `Enumeration`.